pkg/ipnet/ipnet.go

```diff
@@ -27,6 +27,22 @@ func (ipnet *IPNet) String() string {
 	return ipnet.IPNet.String()
 }
 
+// Version returns the IP version for an IPNet.
+func (ipnet *IPNet) Version() int {
```

Comment:
Why? Because it's clearer to read, for both of these functions.

Both examples are improvements to the standard library's IPnet. `To4() == nil` is an awfully unreadable way to check for an IP version.

Reply:
FWIW, other people who have extended IPNet also took similar approaches.  libcalico implements Version(). A few people have implemented CIDR() functions since Size() is more at the lower level running the number of 1's and 0 bits.

But I'm not going to spend time trying to convince you, you seem to already have made up your mind.